### PR TITLE
Bug 1842043: Stop reconciliation on ES CR not present

### PR DESCRIPTION
**Pull request by periklis**
To address: https://bugzilla.redhat.com/show_bug.cgi?id=1842043

Needs backport: release-4.5

### CODE REVIEW

File: pkg/k8shandler/cr.go [hunk 1]
```diff
@@ -2,22 +2,28 @@ package k8shandler
...
+	ErrEsListEmpty  = errors.New("empty elasticsearch list CR")
+)
+
 func GetElasticsearchCR(c client.Client, ns string) (*loggingv1.Elasticsearch, error) {
 	esl := &loggingv1.ElasticsearchList{}
 	opts := &client.ListOptions{Namespace: ns}
 
 	if err := c.List(context.TODO(), opts, esl); err != nil {
-		return nil, fmt.Errorf("failed to find elasticsearch instance in %q: %s", ns, err)
+		return nil, fmt.Errorf("failed to find elasticsearch instance in %q: %w", ns, ErrEsCRNotFound)
```
Comment (igor-karpukhin) [inline pkg/k8shandler/cr.go:22]: `\n` is missing at the end of the format string

File: pkg/k8shandler/cr.go [hunk 2]
```diff
@@ -2,22 +2,28 @@ package k8shandler
...
 	}
 
 	if len(esl.Items) == 0 {
-		return nil, fmt.Errorf("failed to find elasticsearch instance in %q: empty result set", ns)
+		return nil, fmt.Errorf("failed to find elasticsearch instance in %q: %w", ns, ErrEsListEmpty)
```
Comment (igor-karpukhin) [inline pkg/k8shandler/cr.go:26]: same thing as above

File: pkg/controller/proxyconfig/controller.go [hunk 3]
```diff
@@ -82,6 +83,10 @@ type ReconcileProxyConfig struct {
 func (r *ReconcileProxyConfig) Reconcile(request reconcile.Request) (reconcile.Result, error) {
 	es, err := k8shandler.GetElasticsearchCR(r.client, request.Namespace)
 	if err != nil {
+		if errors.Is(err, k8shandler.ErrEsListEmpty) {
```
Comment (ewolinetz) [inline pkg/controller/proxyconfig/controller.go:86]: i believe there is an `errors.IsNotFound` that we can use to short-hand this... no?

File: pkg/k8shandler/cr.go [hunk 4]
```diff
@@ -2,22 +2,28 @@ package k8shandler
...
 
 	loggingv1 "github.com/openshift/elasticsearch-operator/pkg/apis/logging/v1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+var (
+	ErrEsCRNotFound = errors.New("elasticsearch CR instance not found")
+	ErrEsListEmpty  = errors.New("empty elasticsearch list CR")
+)
+
```
Comment (ewolinetz) [inline pkg/k8shandler/cr.go:16]: i dont think we need to do this if we use `errors.IsNotFound`

**Comment by ewolinetz** [comment]
@periklis a few comments to simplify this logic

**Pull request merged**